omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java

```diff
@@ -0,0 +1,102 @@
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+
+@Resource(name = RestConstants.VERSION_1 + IdgenRestController.IDGEN_NAMESPACE
+        + "/viewlogentries", supportedClass = LogEntry.class, supportedOpenmrsVersions = { "1.9.*", "1.10.*", "1.11.*",
+                "1.12.*", "2.0.*", "2.1.*" })
+
+public class LogEntryResource extends DelegatingCrudResource<LogEntry> {
+
+    @Override
+    protected NeedsPaging<LogEntry> doGetAll(RequestContext context) {
+        return new NeedsPaging<LogEntry>(
+                Context.getService(IdentifierSourceService.class).getLogEntries(null, null, null, null, null, null),
+                context);
+    }
+
+    @Override
+    protected PageableResult doSearch(RequestContext context) {
+        IdentifierSourceService identifierSourceService = Context.getService(IdentifierSourceService.class);
+        UserService service = Context.getUserService();
+
+        String source = context.getRequest().getParameter("source");
+        String fromDate = context.getRequest().getParameter("fromDate");
+        String toDate = context.getRequest().getParameter("toDate");
+        String identifier = context.getRequest().getParameter("identifier");
+        String comment = context.getRequest().getParameter("comment");
+        String generatedBy = context.getRequest().getParameter("generatedBy");
+        
+
+        List<LogEntry> logEntries = new ArrayList<LogEntry>();
+
+        IdentifierSource logSource = source != null ? identifierSourceService.getIdentifierSourceByUuid(source) : null;
+        Date dateFrom = fromDate != null ? (Date) ConversionUtil.convert(fromDate, Date.class) : null;
+        Date dateTo = toDate != null ? (Date) ConversionUtil.convert(toDate, Date.class) : null;
+        User user = generatedBy != null ? service.getUserByUuid(generatedBy) : null;
+
+        logEntries = identifierSourceService.getLogEntries(logSource, dateFrom, dateTo, identifier, user, comment);
+        return new NeedsPaging<LogEntry>(logEntries, context);
+    }
+
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+        return null;
+    }
+
+    @Override
+    public LogEntry newDelegate() throws ResourceDoesNotSupportOperationException {
+        return null;
+
+    }
+
+    @Override
+    public LogEntry save(LogEntry delegate) throws ResourceDoesNotSupportOperationException {
+        return null;
```

Comment:
Try to fill more of these methods bodies by copying ideas from that pull request or even the classes i pointed you to in the ticket description.

Reply:
Ive added some more